experimental/iree/compile_workloads.py

```diff
@@ -0,0 +1,231 @@
+  parser.add_argument("-o",
+                      "--output",
+                      dest="output_dir",
+                      type=pathlib.Path,
+                      required=True,
+                      help="Directory where output artifacts are saved.")
+  parser.add_argument("-name",
+                      "--benchmark_name",
+                      type=str,
+                      required=True,
+                      help="The regex pattern to match benchmark names.")
+  parser.add_argument("-device",
+                      "--target_device",
+                      dest="target_device_name",
+                      type=str,
+                      required=True,
+                      choices=ALL_DEVICE_NAMES,
+                      help="The target device to benchmark.")
+  parser.add_argument("--iree-compile-path",
+                      "--iree_compile_path",
+                      type=pathlib.Path,
+                      required=True,
+                      help="Path to the IREE Compiler tool.")
+  parser.add_argument("--temp-dir",
+                      "--temp_dir",
+                      type=pathlib.Path,
+                      default=pathlib.Path("/tmp/openxla-benchmark"),
+                      help="Temporary directory where intermediates are saved.")
+  parser.add_argument("--no-download",
+                      "--no_download",
+                      action="store_true",
+                      help="Don't automatically download benchmark artifacts.")
+  parser.add_argument("--verbose",
+                      action="store_true",
+                      help="Show verbose messages.")
+  return parser.parse_args()
+
+
+def _download_artifacts(benchmarks: Sequence[def_types.BenchmarkCase],
+                        temp_dir: pathlib.Path,
+                        verbose: bool = False):
+  """Download benchmark artifacts."""
+
+  download_list = []
+  for benchmark in benchmarks:
+    model = benchmark.model
+    artifacts_dir_url = model.artifacts_dir_url
+    if artifacts_dir_url is None:
+      raise ValueError(f"Artifacts dir URL is not found in '{model.name}'.")
+
+    if def_types.ModelArtifactType.STABLEHLO_MLIR in model.exported_model_types:
+      mlir_path = temp_dir / model.name / "stablehlo.mlirbc"
+      mlir_url = artifacts_dir_url + "/stablehlo.mlirbc"
+      download_list.append((mlir_url, mlir_path))
+
+    if def_types.ModelArtifactType.TOSA_MLIR in model.exported_model_types:
+      mlir_path = temp_dir / model.name / "tosa.mlirbc"
+      mlir_url = artifacts_dir_url + "/tosa.mlirbc"
+      download_list.append((mlir_url, mlir_path))
+
+    input_path = temp_dir / model.name / "inputs_npy.tgz"
+    input_url = artifacts_dir_url + "/inputs_npy.tgz"
+    download_list.append((input_url, input_path))
+
+    expect_path = temp_dir / model.name / "outputs_npy.tgz"
+    expect_url = artifacts_dir_url + "/outputs_npy.tgz"
+    download_list.append((expect_url, expect_path))
```

Comment:
Do we need to download input/output when compiling the models?

Reply:
We don't but I wanted to be able to create a compiled artifacts directory with everything needed to run benchmarks inside of it. Then it would be a simple adb push of the directory on the host.